### PR TITLE
Add guide on Poetry

### PR DESCRIPTION
**Pull request by astrojuanlu**
Clarifies how Poetry is supported on Read the Docs.

### CODE REVIEW

**Review by agjohnson — APPROVED** [review]
Looks great! There maybe there isn't really anything substantial here, but maybe we want to mention how to verify that the dependencies are being used? This guide goes to mention pinning and the lock file, but might leave the reader guessing how to wrap this process up.

File: docs/guides/poetry.rst [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+Declaring your project metadata
+-------------------------------
+
+Poetry is a :pep:`517`-compliant build backend, which means that
+`it generates your project
+metadata <https://python-poetry.org/docs/pyproject/#poetry-and-pep-517>`_
+using a standardized interface that can be consumed directly by pip.
+Therefore, by making sure that
+the ``build-system`` section of your ``pyproject.toml``
+declares the build backend as follows:
```
[the remark this answers is not in the document]
Comment (astrojuanlu) [reply docs/guides/poetry.rst:13]: Good point, I will rephrase this part

**Comment by astrojuanlu** [comment]
:robot: Rendered version: https://docs--8702.org.readthedocs.build/en/8702/guides/poetry.html

**Comment by astrojuanlu** [comment]
> There maybe there isn't really anything substantial here,

Yep, it ended up being quite short. At least now "poetry" returns something in the search.

> but maybe we want to mention how to verify that the dependencies are being used? This guide goes to mention pinning and the lock file, but might leave the reader guessing how to wrap this process up.

Opened #8704 to track this discussion in a more general way, so it's useful for other pages as well.

**Pull request merged**